pkg/upgrade/upgrader/cstor_cspi.go

```diff
@@ -295,3 +309,90 @@ func (obj *CSPIPatch) verifyCSPIVersionReconcile() (string, error) {
 	}
 	return "", nil
 }
+
+func (obj *CSPIPatch) upgradeBackupRestore() (string, error) {
+	// Migrate backup to v1 version
+	oldBackupList, err := obj.OpenebsClientset.OpenebsV1alpha1().
+		CStorBackups(obj.OpenebsNamespace).List(metav1.ListOptions{
+		LabelSelector: types.CStorPoolInstanceNameLabelKey + "=" + obj.Name,
+	})
+	if err != nil {
```

Comment:
Let's make a check like 
```go
if err != nil && !k8serror.IsNotFound(err) {
}
```
So If any user-installed 2.2 version of cstor-operator he will not have v1alpha1 backup CRD in that case error will be returned from here and user can't upgrade to further versions. 

The same check can be applied for other resources and deletion also.